### ISSUE TITLE
Combine emotion stylesheets into a single stylesheet during SSR

### ISSUE THREAD

**Issue by davidflanagan** [mdn/sprints#1654]
# User story
As an MDN developer, I want to tweak the SSR process so that the Emotion stylesheets are coalesced into a single stylesheet, to improve performance.

# Acceptance criteria

- View source on a beta page shows that there is only one emotion-generated stylesheet in the HTML, rather than the current situation where there are many &lt;style data-emotion-css&gt; elements.

**Comment by davidflanagan** [comment]
Estimating 1 point for this issue, because I'm hoping that we can solve it either by changing a setting in a config file somewhere or by writing some relatively simple search-and-replace string manipulation code to post-process the output of React's renderToString() function.

**Comment by davidflanagan** [comment]
I actually attempted to do this, and it turned out to break SSR. Emotion generates separate stylesheets, and if we combine them as part of the SSR process, then the hydration step fails because the HTML does not match what React expects, and we end up doing a re-render after the page load, and are worse off than we would be.

I left the code for my attempt commented out at the bottom of kuma/javascript/src/ssr.jsx in case anyone wants to try to revive it.

Note however that my original estimate of 1 point is now way off, and the way forward here might be a feature request or a PR to the Emotion repo rather than something we can fix.

Also @estelle may know more about this, but I'm not aware of any hard data about the perf impact of having lots of little stylesheets embedded in a just-in-time fashion in the HTML file versus having one big stylesheet up at the top of the HTML file. I can imagine that the way emotion does things with little stylesheets throughout the document might result in better above-the-fold rendering times.

Again, I've got no data here, but fixing this isn't easy, and it might not be much of a perf improvement.  Before working on this, I'd spend time working to eliminate CSS that we no longer need on the React site from the big SASS-derived stylesheets.
